src/server/terrain/structure_building_blocks.zig

```diff
@@ -101,6 +102,100 @@ pub fn isOriginBlock(block: Block) bool {
+		};
+		const self = StructureBuildingBlock{
+			.children = arenaAllocator.alloc(AliasTable(Child), childBlockStringId.items.len),
+			.blueprints = blueprints,
+		};
+		const childrenZon = zon.getChild("children");
+		for(childBlockStringId.items, 0..) |colorName, colorIndex| {
+			self.children[colorIndex] = try initChildTableFromZon(stringId, colorName, colorIndex, childrenZon.getChild(colorName));
+		}
+		return self;
+	}
+	pub fn getBlueprint(self: StructureBuildingBlock, rotation: Degrees) *BlueprintEntry {
+		return &self.blueprints[@intFromEnum(rotation)];
+	}
+	pub fn pickChild(self: StructureBuildingBlock, block: BlueprintEntry.StructureBlock, seed: *u64) *const StructureBuildingBlock {
+		return self.children[block.index].sample(seed).structure;
+	}
+};
+
+fn initChildTableFromZon(parentId: []const u8, colorName: []const u8, colorIndex: usize, zon: ZonElement) !AliasTable(Child) {
+	if(zon == .null) return .init(arenaAllocator, &.{});
+	if(zon != .array) {
+		std.log.err("['{s}'->'{s}'] Incorrect child data structure, array expected.", .{parentId, colorName});
+		return .init(arenaAllocator, &.{});
+	}
+	if(zon.array.items.len == 0) {
+		std.log.warn("['{s}'->'{s}'] Empty children list.", .{parentId, colorName});
+		return .init(arenaAllocator, &.{});
+	}
+	const list = arenaAllocator.alloc(Child, zon.array.items.len);
+	for(zon.array.items, 0..) |entry, childIndex| {
+		list[childIndex] = try Child.initFromZon(parentId, colorName, colorIndex, childIndex, entry);
+	}
+	return .init(arenaAllocator, list);
+}
+
+const Child = struct {
+	structure: *StructureBuildingBlock,
+	chance: f32,
+
+	fn initFromZon(parentId: []const u8, colorName: []const u8, colorIndex: usize, childIndex: usize, zon: ZonElement) !Child {
+		const structureId = zon.get([]const u8, "structure", "");
+		if(structureId.len == 0) {
+			std.log.err("['{s}'->'{s}'->'{d}'] Child node has empty structure field, parent structure will be discarded.", .{parentId, colorName, childIndex});
+			return error.EmptyStructureId;
+		}
+		childrenToResolve.append(.{.parentId = parentId, .colorName = colorName, .colorIndex = colorIndex, .childIndex = childIndex, .structureId = structureId});
+		return .{
+			.structure = undefined,
+			.chance = zon.get(f32, "chance", 1.0),
+		};
+	}
+};
+
+pub fn registerSBB(structures: *std.StringHashMap(ZonElement)) !void {
+	std.debug.assert(structureCache.capacity() == 0);
+	structureCache.ensureTotalCapacity(arenaAllocator.allocator, structures.count()) catch unreachable;
+	childrenToResolve = .init(main.stackAllocator);
+	{
+		var iterator = structures.iterator();
+		while(iterator.next()) |entry| {
+			const value = StructureBuildingBlock.initFromZon(entry.key_ptr.*, entry.value_ptr.*) catch continue;
+			const key = arenaAllocator.dupe(u8, entry.key_ptr.*);
+			structureCache.put(arenaAllocator.allocator, key, value) catch unreachable;
+			std.log.debug("Registered structure building block: '{s}'", .{entry.key_ptr.*});
```

Comment:
I think it would be better to put this log at the start of the loop, so we can see the name of the structure that failed in the log.

Reply:
I would rather opt for proper error message after initFromZon

Reply:
Yeah that also works I guess.